workflow/client.go

```diff
@@ -0,0 +1,174 @@
+//go:generate counterfeiter ./ Client
+
+package workflow
+
+import (
+	// "encoding/json"
+	// "errors"
+	// "fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/3dsim/auth0"
+	"github.com/3dsim/workflow-goclient/genclient"
+	"github.com/3dsim/workflow-goclient/genclient/operations"
+	"github.com/3dsim/workflow-goclient/models"
+	"github.com/PuerkitoBio/rehttp"
+	openapiclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	// "github.com/go-openapi/swag"
+	log "github.com/inconshreveable/log15"
+)
+
+// Log is a github.com/inconshreveable/log15.Logger.  Log is exposed so that users of this library can set
+// their own log handler.  By default this Log uses the DiscardHandler, which discards log statements.
+// See: https://godoc.org/github.com/inconshreveable/log15#hdr-Library_Use
+//
+// To set a different log handler do something like this:
+//
+// 		Log.SetHandler(log.LvlFilterHandler(log.LvlInfo, log.CallerFileHandler(log.StdoutHandler)))
+var Log = log.New()
+
+func init() {
+	Log.SetHandler(log.DiscardHandler())
+}
+
+const (
+	// WorkflowAPIBasePath is the base path or "slug" for the workflow api
+	WorkflowAPIBasePath = "workflow-api"
+)
+
+// Client is a wrapper around the generated client found in the "genclient" package.  It provides convenience methods
+// for common operations.  If the operation needed is not found in Client, use the "genclient" package using this client
+// as an example of how to utilize the genclient.  PRs are welcome if more functionality is wanted in this client package.
+type Client interface {
+	GetWorkflow(workflowID string) (*models.Workflow, error)
```

Comment:
It's a weird idiom in Go, but they don't typically use `Get` for getters.  https://golang.org/doc/effective_go.html#Getters.  Of course... this isn't technically a `getter` so not sure of the best practice.  But in the other clients we've just been using `Entity` instead of `GetEntity`